lib/vector-core/src/event/mod.rs

```diff
@@ -79,6 +85,11 @@ impl Event {
         Event::Log(LogEvent::default())
     }
 
+    #[must_use]
+    pub fn new_empty_trace() -> Self {
+        Event::Trace(TraceEvent::default())
+    }
+
```

Comment:
I think we'd like to discourage the use of an interface like this in favour of `TraceEvent::default().into()`